browser/ui/webui/brave_education/education_page_ui.cc

```diff
@@ -0,0 +1,119 @@
+
+#include "brave/browser/ui/webui/brave_education/education_page_ui.h"
+
+#include <string>
+#include <utility>
+
+#include "brave/browser/ui/webui/brave_browser_command/brave_browser_command_handler.h"
+#include "brave/browser/ui/webui/brave_education/brave_education_handler.h"
+#include "brave/browser/ui/webui/brave_education/brave_education_page_delegate_desktop.h"
+#include "brave/browser/ui/webui/brave_webui_source.h"
+#include "brave/components/brave_education/education_urls.h"
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/grit/brave_education_resources.h"
+#include "brave/grit/brave_education_resources_map.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/tabs/public/tab_interface.h"
+#include "chrome/browser/ui/webui/webui_util.h"
+#include "chrome/grit/branded_strings.h"
+#include "chrome/grit/generated_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "services/network/public/mojom/content_security_policy.mojom.h"
+
+namespace {
+
+void CreateAndAddWhatsNewUIHtmlSource(content::WebUI* web_ui,
+                                      Profile* profile) {
+  content::WebUIDataSource* source =
+      content::WebUIDataSource::CreateAndAdd(profile, kBraveGettingStartedHost);
+  webui::SetupWebUIDataSource(
+      source, base::span<const webui::ResourcePath>(kBraveEducationResources),
+      IDR_BRAVE_EDUCATION_BRAVE_EDUCATION_HTML);
+
+  AddBackgroundColorToSource(source, web_ui->GetWebContents());
+
+  static constexpr webui::LocalizedString kStrings[] = {
+      {"title", IDS_WELCOME_HEADER}};
+  source->AddLocalizedStrings(kStrings);
+
+  // Allow embedding of iframe content from allowed domains.
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ChildSrc,
+      "child-src chrome://webui-test https://brave.com/ "
+      "https://2263.pr.bravesoftware.com;");
+}
+
+}  // namespace
+
+EducationPageUI::EducationPageUI(content::WebUI* web_ui, const GURL& url)
+    : ui::MojoWebUIController(web_ui, /*enable_chrome_send=*/true),
+      page_factory_receiver_(this),
+      browser_command_factory_receiver_(this),
+      profile_(Profile::FromWebUI(web_ui)) {
+  CreateAndAddWhatsNewUIHtmlSource(web_ui, profile_);
+}
+
+WEB_UI_CONTROLLER_TYPE_IMPL(EducationPageUI)
+
+void EducationPageUI::BindInterface(
+    mojo::PendingReceiver<brave_education::mojom::PageHandlerFactory>
+        receiver) {
+  page_factory_receiver_.reset();
+  page_factory_receiver_.Bind(std::move(receiver));
+}
+
+void EducationPageUI::CreatePageHandler(
+    mojo::PendingRemote<brave_education::mojom::Page> page,
+    mojo::PendingReceiver<brave_education::mojom::PageHandler> receiver) {
+  DCHECK(page);
+
+  auto* web_contents = web_ui()->GetWebContents();
+  auto* tab = tabs::TabInterface::GetFromContents(web_contents);
```

Comment:
`GetFromContents` is only meant for transitioning old code and doesn't seem to serve any purpose here.

Reply:
Fixed in https://github.com/brave/brave-core/pull/26154/commits/b8754b2a9bd54be5fe0c2216ea7684b5bd523bad